src/classes/Entity/types.ts

```diff
@@ -437,39 +438,42 @@ export interface UpdateCustomParameters {
 
 export type UpdateCustomParams = O.Partial<UpdateCustomParameters & DocumentClient.UpdateItemInput>
 
-export type UpdateItem<
-  MethodItemOverlay extends Overlay,
+export type UpdateItem<MethodItemOverlay extends Overlay,
   EntityItemOverlay extends Overlay,
   CompositePrimaryKey extends O.Object,
   Item extends O.Object,
   Attributes extends ParsedAttributes,
-  StrictSchemaCheck extends boolean | undefined = true
-> = FirstDefined<
-  [
+  StrictSchemaCheck extends boolean | undefined = true> = FirstDefined<[
     MethodItemOverlay,
     EntityItemOverlay,
-    A.Compute<
-      CompositePrimaryKey &
-        {
-          [inputAttr in Attributes['always']['input']]:
-            | Item[A.Cast<inputAttr, keyof Item>]
-            | { $delete?: string[]; $add?: any; $prepend?: any[]; $append?: any[] }
-        } &
-        {
-          [optAttr in Attributes['required']['all'] | Attributes['always']['default']]?:
-            | Item[A.Cast<optAttr, keyof Item>]
-            | { $delete?: string[]; $add?: any; $prepend?: any[]; $append?: any[] }
-        } &
-        {
-          [attr in Attributes['optional']]?:
-            | null
-            | Item[A.Cast<attr, keyof Item>]
-            | { $delete?: string[]; $add?: any; $append?: any[]; $prepend?: any[] }
-        } & { $remove?: Attributes['optional'] | Attributes['optional'][] }
-    >
+    A.Compute<CompositePrimaryKey &
+      {
+        [inputAttr in Attributes['always']['input']]:
+        | If<A.Equals<Item[A.Cast<inputAttr, keyof Item>], FromDynamoData<'list' | 'set'> | undefined>, { $delete?: string[]; $add?: any; $prepend?: Item[A.Cast<inputAttr, keyof Item>]; $append?: Item[A.Cast<inputAttr, keyof Item>]; $remove?: number[] } | Item[A.Cast<inputAttr, keyof Item>], Item[A.Cast<inputAttr, keyof Item>]>
+        | If<A.Equals<Item[A.Cast<inputAttr, keyof Item>], number[] | undefined>, { $delete?: number[]; $add?: number[]; $prepend?: Item[A.Cast<inputAttr, keyof Item>]; $append?: number[]; } | string[]>
```

Comment:
I think you can factorize in a `SetOrListUpdateInput<T extends string | number | boolean> = { $delete?: T[], $add?:any, $append?: ... }`